controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -327,6 +328,35 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	}
 
 	if cluster.Status().State() != cmv1.ClusterStateUninstalling {
+		machinePools, err := rosaScope.GetMachinePools()
+		if err != nil {
+			rosaScope.Info("Error getting controlPlan machinePools ", err)
+			return ctrl.Result{}, err
+		}
+
+		var errs []error
+		for id, mp := range machinePools {
+			if mp.DeletionTimestamp != nil && !mp.DeletionTimestamp.IsZero() {
+				continue
+			}
+			err = rosaScope.Client.Delete(context.TODO(), &machinePools[id])
+			if err != nil {
+				rosaScope.Info("Error deleting machinePool ", err)
+				errs = append(errs, err)
+				continue
+			}
```

Comment:
```suggestion
			if err := rosaScope.Client.Delete(ctx, &machinePools[id]); err != nil {
				errs = append(errs, err)
			}
```
why log an error as info? we are returning the errors, so they will be logged anyway.
Also, there is no need to `continue` here.
Please always use the provided context over `context.TODO()`